rfc/2022/2022-07-20 - Update Build Directory Structure of vets-website.md

```diff
@@ -0,0 +1,61 @@
+
+Storing isolated application bundles with bundles and assets from full builds also contributes to a race condition between continuous production deployments of isolated applications and daily production deployments. In the planned implementation of continuous production deployments, isolated applications will be deployed to production each time a commit with changes to the application(s) is merged into the `main` branch. If a branch with changes to an isolated application is merged into `main` during the time of the daily production deployment, the continuous production deployment triggered by the merge could end up writing to the isolated application's bundles in the S3 bucket at the same time as the daily production deploy, assuming the daily production deploy continued building and deploying all applications. Even if isolated applications were excluded from the daily production deployment, based on the current directory structure, changes to the global bundles and assets in the daily production deploy could break the untouched isolated application bundles in the S3 deployment buckets, inversely creating the same problem mentioned earlier.
+
+Separating bundles and assets of isolated application builds from bundles and assets of full builds can prevent potentially breaking isolated applications in production.
+
+## Motivation
+
+Storing the bundles and assets of both build types separately would resolve issues that can occur from syncing isolated application bundles with global bundles and assets generated from previous deployments.
+
+## Design
+
+### Updating the structure of the `build` directory
+
+Restructuring the `build/generated` directory involves updating the `vets-website` build to not store bundles of isolated applications in the root of the directory. Doing so means that the root directory will only contain bundles of applications that aren't isolated, along with the global bundles and assets generated from full builds, or in this case, builds of non-isolated applications. Bundles and assets from isolated application builds will be stored in their own subdirectory within the `build/generated` directory, e.g. `build/generated/check-in`, `build/generated/vaos`, etc. Doing this will ensure two things:
+
+1. Bundles and assets from isolated application builds will be isolated from other builds.
+2. Isolated application builds can be deployed independently of *full builds* (non-isolated application builds).
+
+Updating the `build/generated` directory to follow this structure means that *full builds* will need to exclude the entry files of isolated applications, which ultimately excludes their bundles from being generated. The Webpack config will need to be updated to account for this, along with knowing when and where to store bundles and assets from isolated application builds.
```

Comment:
They would use their own versions of previously global assets/bundles that get generated in their build. So the `web-components.entry.js`, `web-components.css`, etc. files that get generated in an isolated app build would be used by that app's page on the site. Non-isolated apps would rely on the same global assets/bundles as all apps currently do right now, since non-isolated apps would get built and deployed together